library/x509_crt.c

```diff
@@ -1473,6 +1756,104 @@ static int x509_info_subject_alt_name( char **buf, size_t *size,
     return( 0 );
 }
 
+int mbedtls_x509_parse_subject_alternative_name( const mbedtls_x509_crt *crt,
+                                                 mbedtls_x509_subject_alternative_name **san )
+{
+    int ret;
+    const mbedtls_x509_sequence *cur = &crt->subject_alt_names;
+    mbedtls_x509_subject_alternative_name *cur_san = *san, *prev_san = NULL;
+
+    if( crt->ext_types & MBEDTLS_X509_EXT_SUBJECT_ALT_NAME )
+    {
+        if( cur_san != NULL )
+        {
+            return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+        }
+
+        while( cur != NULL )
+        {
+            switch( cur->buf.tag &
+                ( MBEDTLS_ASN1_TAG_CLASS_MASK | MBEDTLS_ASN1_TAG_VALUE_MASK ) )
+            {
+                /*
+                 * otherName
+                 */
+                case( MBEDTLS_ASN1_CONTEXT_SPECIFIC | MBEDTLS_X509_SAN_OTHER_NAME ):
+                {
+                    mbedtls_x509_san_other_name other_name;
+
+                    ret = x509_get_other_name( cur, &other_name );
+                    if( ret != 0 )
+                    {
+                        /*
+                         * In case MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE is returned,
+                         * then the "otherName" is of an unsupported type. Ignore.
+                         */
+                        if( ret == MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE )
+                            ret = 0;
+                        continue;
```

Comment:
I am a bit unsure about this. If you have something that isnt supported in the list, you execute `continue` but I think that you may not make `cur` point to the next item on the list. This is because:
```
            cur = cur->next;
```
is only after the end of the switch. So does this go into some kind of infinite loop when there is an unsupported otherName in the list? Do we have a test for this?

Reply:
I believe you are correct. This also means that there is a missing test. I'll add a certificate with a different unsupported `otherName`.